Tests/SourceKitTests/BuildSystemTests.swift

```diff
@@ -227,6 +227,107 @@ final class BuildSystemTests: XCTestCase {
     }
   }
 
+  func testClangdDocumentFallbackWithholdsDiagnostics() {
+    guard haveClangd else { return }
+
+    let url = URL(fileURLWithPath: "/\(#function)/file.m")
+    let doc = DocumentURI(url)
+    let args = [url.path, "-DDEBUG"]
+    let text = """
+    #ifdef FOO
+    static void foo() {}
+    #endif
+
+    int main() {
+      foo();
+      return 0;
+    }
+    """
```

Comment:
Style: please indent the multi-line string one level.

Reply:
Done although other tests in the file don't abide by the format